### ISSUE TITLE
Provisioning machine job can't finish

### ISSUE THREAD

**Issue by FrankYang0529** [rancher/rancher#36547]
**Rancher Server Setup**
- Rancher version: v2.6.3
- Installation option (Docker install/Helm Chart): Docker

**Information about the Cluster**
- Kubernetes version: v1.22.6+rke2r1
- Cluster Type (Local/Downstream): Infrastructure Provider - Harvester

**User Information**
- What is the role of the user logged in? Admin
   
**Describe the bug**
Try to add node to a RKE2 cluster on Harvester. The Job may get deleted and have OOM issue with following logs:
```
Error creating machine: Error saving host to store after attempting creation: unable to load secret : error getting secret from kubernetes: Unauthorized
The default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag.
```
Please see details in this thread https://github.com/harvester/harvester/issues/1823

**To Reproduce**
1. Create an RKE2 cluster with image JeOS on Harvester.
2. When provisioning is finished, add a node to the cluster.

**Result**
The new provisioning job may fail and the machine is stuck in `Provisioning` status.

**Expected Result**
The machine can be created successfully.

**Additional context**
https://github.com/harvester/harvester/issues/1823

**Comment by TachunLin** [comment]
The issue is not yet been fixed according to the related issue comment: https://github.com/harvester/harvester/issues/1823#issuecomment-1049501074

**Comment by TachunLin** [comment]
Verified **fixed** on master-088a020f-head. (03/06) with Rancher `v2.6-head`. (`33db2f5`). Close this issue.

 ### Result
* We can scale pool up by clicking + to add node to RKE2 cluster by node driver
![image](https://user-images.githubusercontent.com/29251855/157243631-19d64790-f726-41ca-83f9-80740852035c.png)
![image](https://user-images.githubusercontent.com/29251855/157244484-05552ad7-9b3d-497c-9b97-2c6e9de1b5ce.png)
![image](https://user-images.githubusercontent.com/29251855/157244555-92ab6fdb-2f34-402f-9570-1659006f0bf1.png)

* Can also click - to scale pool down by deleting a node from existing cluster 
![image](https://user-images.githubusercontent.com/29251855/157249545-1a264946-4651-4643-b182-479f6574588d.png)


### Test Information
* Test Environment: 1 node1 harvester on local kvm machine
* Harvester version: master-088a020f-head. (03/06)
* Rancher version: v2.6-head. (33db2f5)
* Kubernetes version: `v1.21.10+rke2r1`
* OS image: `SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-GM.qcow2`

### Verify Steps
1. Upload `SLES15-SP3-JeOS.x86_64-15.3-OpenStack-Cloud-GM.qcow2` image to harvester cluster
2. Import harvester in rancher
3. Create cloud credential
4. Provision a RKE2 cluster 
5. Select RKE2 version: `v1.21.10+rke2r1`
6. Add the following cloud init in advance settings according to the workaround 
https://github.com/harvester/harvester/issues/1973#issuecomment-1061604853 
  ```
  #cloud-config
  runcmd:
  - zypper ar -cfG 'https://mirrors.tuna.tsinghua.edu.cn/opensuse/tumbleweed/repo/oss/' tuna-oss
  - zypper ar -cfG 'https://mirrors.tuna.tsinghua.edu.cn/opensuse/tumbleweed/repo/non-oss/' tuna-non-oss
  - zypper ref
  - zypper -n in glibc-2.35-2.1.x86_64 glibc-locale-base-2.35-2.1.x86_64 filesystem-84.87-3.2.x86_64 apparmor-parser-3.0.4-1.2.x86_64
  ```
7. Click + to scale pool up by adding node after provisioning RKE2 complete,
  ![image](https://user-images.githubusercontent.com/29251855/157246863-d31517eb-df23-4962-8e60-3cd97d50d5dc.png)

8. Check can add node correctly to existing rke2 cluster 
9. Click - to scale pool down by deleting a node from existing cluster

**Comment by slickwarren** [comment]
Also tested on rancher v2.6-head (76f5fb6) using v1.21.10 for downstream clusters:
* deploy all-roles single node cluster - scale up nodes -- pass
  * scale down -- pass
* deploy single node per role cluster - scale up each pool individually -- pass
  * scale down each individually -- pass
* deploy mixed roles cluster - scale up nodes -- pass
  * scale down -- pass
* manually add a new node to an existing cluster (cluster -> edit -> add node) -- pass
